cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java

```diff
@@ -86,14 +105,16 @@ public DatasetInstanceService(DatasetTypeManager typeManager, DatasetInstanceMan
     this.metaCache = CacheBuilder.newBuilder().build(
       new CacheLoader<Id.DatasetInstance, DatasetMeta>() {
         @Override
-        public DatasetMeta load(Id.DatasetInstance datasetId) throws Exception {
+        public DatasetMeta load(@SuppressWarnings("NullableProblems") Id.DatasetInstance datasetId) throws Exception {
           return getFromMds(datasetId);
         }
       }
     );
+    this.authorizationEnforcementService = authorizationEnforcementService;
+    this.authorizerInstantiator = authorizerInstantiator;
   }
 
-  @SuppressWarnings("unused")
+  @VisibleForTesting
   @Inject(optional = true)
   public void setAuditPublisher(AuditPublisher auditPublisher) {
```

Comment:
Is the test not in the same package? If yes, can this method be package-private?